Code/EnginePlugins/AngelScriptPlugin/TypeRegistration/AsRegister_Reflected.cpp

```diff
@@ -491,13 +491,8 @@ void ezAngelScriptEngineSingleton::RegisterSingleGenericFunction(const char* szF
 
         if (defaultValue.IsValid())
         {
-          if (pArgType->GetVariantType() == ezVariantType::UInt32)
-          {
-            // special case to make the default value unsigned
-            defaultValue = defaultValue.ConvertTo<ezUInt32>();
-          }
-
-          decl.Append(ezAngelScriptUtils::DefaultValueToString(defaultValue));
+          ezVariantType::Enum expectedType = bIsEnum ? ezVariantType::Int64 : pArgType->GetVariantType();
```

Comment:
no idea what this change is about

Reply:
The value in ezDefaultValueAttribute may not have the correct type for a function parameter. Thus we pass the type of the function parameter and convert the value to that type. Otherwise you can end up with cases like bool b = 1 in the function declaration und angel script does not like that.